pkg/bundle/bundle.go

```diff
@@ -86,6 +87,11 @@ func (b *ProtobufBundle) validate() error {
 		if len(entries) > 0 && !b.hasInclusionProof {
 			return errors.New("inclusion proof missing in bundle (required for bundle v0.2)")
 		}
+	case SigstoreBundleMediaType03:
+		cert := b.Bundle.VerificationMaterial.GetCertificate()
+		if cert == nil {
+			return errors.New("verification material must be single X.509 certificate (required for bundle v0.3)")
```

Comment:
This is too strict of a check, this wouldn't work for private deployments. It's only required when the instance is the public good instance - https://github.com/sigstore/protobuf-specs/blob/main/protos/sigstore_bundle.proto#L91-L92

Reply:
Oops, I didn't mean to leave out folks using their own key!

So one option is we could say for bundle v0.3 you have to either have a certificate or a key (but not a certificate chain).

Another more literal option would be to go a step further, see if the certificate claims to be from the public good (and public staging?) instance, and only perform the check listed above if we're on PGI. Do we need to be that specific? That is, do we want to support private Sigstore deployments issuing v0.3 bundles using the keyless flow that put the certificate in a certificate chain?

Reply:
I pushed a change assuming we don't actually want to check if it's PGI or not, but if we do need to support private Sigstore deployments issuing v0.3 bundles with intermediates in a certificate chain let me know.

Reply:
Given that we didn't have support for intermediates in the chain before, this change looks good. We can relax this later.

> That is, do we want to support private Sigstore deployments issuing v0.3 bundles using the keyless flow that put the certificate in a certificate chain?

Long-term, yeah, since we don't know how private PKIs will be structured and intermediates may just be seen as chain builders. We might have to control this via a policy flag since there isn't a good way to distinguish between PGI and non-PGI and non-Sigstore PKI. The policy could state whether intermediates in the bundle should be allowed or trusted.